### PR TITLE
94X Update phase1 pixel thresholds

### PR DESCRIPTION
**Pull request by jkarancs**
Threshold and resolution measurements in data suggested the need to increase the simulated pixel threshold values for Layer 1 and 2 for the new phase 1 detector.
This PR contains:
* Updated L1 thresholds: increase from 2000 to 3000 electrons
* Updated L2 thresholds: increase from 2000 to 2600 electrons
* Also implement the missing configuration parameter for the latter

The expected impact on physics will be a slight worsening of pixel related tracking quantities which will better match what we see now in data.

The PR will need to go together with updated conditions for Templates (and likely also Lorentz Angle), which we are preparing now.

Validation results based on data and simulation (including this PR) are attached in a pdf below.

[2017_10_06_ThresholdUpdate.pdf](https://github.com/cms-sw/cmssw/files/1362365/2017_10_06_ThresholdUpdate.pdf)

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
hi @jkarancs @makortel (@slava77 @perrotta) - from a quick review I thought I'd confirm that the loss in tracks indeed is at the level expected. There also may be a large change in +eta forward tracks, which seems to have a larger change than the -eta forward region. Perhaps a fluctuation but perhaps not?

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_4_X_2017-10-07-1100+20775/22974/validateJR/all_OldVSNew_TTbar13TeVPU2017wf10224p0/all_OldVSNew_TTbar13TeVPU2017wf10224p0c_recoTracks_generalTracks__RECO_obj_eta.png

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_9_4_X_2017-10-07-1100+20775/22974/validateJR/all_OldVSNew_TTbar13TeVPU2017wf10224p0/all_OldVSNew_TTbar13TeVPU2017wf10224p0c_log10recoTracks_generalTracks__RECO_obj_pt.png

**Comment by makortel** [comment]
@davidlange6 I'm running a high-stat test to check the effect.

File: SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc [hunk 1]
```diff
@@ -699,16 +703,24 @@ void SiPixelDigitizerAlgorithm::digitize(const PixelGeomDetUnit* pixdet,
...
+	  } else {
+	    thePixelThresholdInE = CLHEP::RandGaussQ::shoot(engine, theThresholdInE_BPix , theThresholdSmearing_BPix); // gaussian smearing
+	  }
 	}
       } else {
-	if((pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)  && lay==1) {
-	  thePixelThresholdInE = theThresholdInE_BPix_L1;
-	} else {
-	  thePixelThresholdInE = theThresholdInE_BPix; // no smearing
+	if(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB) {
```
Comment (perrotta) [inline SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc]: Just a curiosity appeared while looking at the code changes...

With the previous code, if
!(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || 
            pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
thePixelThresholdInE would have been set = theThresholdInE_BPix

With the current code if
!(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel || 
            pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
thePixelThresholdInE remains 0.

My question: can 
(pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel ||
 pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
be false?
  - if the answer is "yes", please check the logic if actually corresponds to what you want;
  - if the answer is "no", why do you bother with that check?

Comment (makortel) [reply SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc]: > can
> (pixdet->subDetector()==GeomDetEnumerators::SubDetector::PixelBarrel ||
> pixdet->subDetector()==GeomDetEnumerators::SubDetector::P1PXB)
> be false?

I'd say yes: given that the code is within
```cpp
 if(pixdet->type().isTrackerPixel() && pixdet->type().isBarrel()){ // Barrel modules
```
the `subDetector()` can be `P2PXB`.

Comment (perrotta) [reply SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc]: Indeed... Although I did not see any change in the Phase2 workflows outputs in jenkins, that's why I'm asking if there is anything that prevents Phase2 code to access this method: I did not check that deep in CMSSW to verify it,

Comment (makortel) [reply SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc]: Let me add some phase2 experts @boudoul @atricomi @delaere

Comment (boudoul) [reply SimTracker/SiPixelDigitizer/plugins/SiPixelDigitizerAlgorithm.cc]: hi @perrotta and @makortel , for phase2 a difference digitizer is called thanks to the phase2 Era : https://github.com/cms-sw/cmssw/blob/master/SimGeneral/MixingModule/python/pixelDigitizer_cfi.py#L14

**Comment by makortel** [comment]
I have here plots for 1000 events of ttbar+PU35 in 940pre2
https://cms-tracking-validation.web.cern.ch/cms-tracking-validation/PR/CMSSW_9_4_0_pre2_PR20775/
On global scale there are little changes (esp. efficiency is the same, fake rate increases by few percent
https://cms-tracking-validation.web.cern.ch/cms-tracking-validation/PR/CMSSW_9_4_0_pre2_PR20775/plots_ootb/effandfakePtEtaPhi.pdf
, and there is tiny degradation of resolutions above 1 GeV
https://cms-tracking-validation.web.cern.ch/cms-tracking-validation/PR/CMSSW_9_4_0_pre2_PR20775/plots_ootb/resolutionsPt.pdf
)

**Comment by davidlange6** [comment]
+1

**Pull request merged**